### PR TITLE
[chore]: bump minimum deployment & swift versions

### PR DESCRIPTION
**Pull request by jamieQ**
### Issue

- we'd like to bump the minimum deployment target and swift version to facilitate development going forward

### Description

- bump minimum iOS deployment target to 14.0
- bump swift version to 5.7
- remove an unneeded config in our gemfile
- bump xcode version in CI configs to 14.2.0
- bump simulator target in CI configs to 16.2
- add `ENABLE_TESTING_SEARCH_PATHS` to WorkflowTesting podspec to resolve search path issues induced by deployment target change
- remove `VERSION` file in favor of `version.rb`, and standardize version references

note: this change will require a major version bump for the next release that includes it

## Checklist

- [x] Unit Tests
- [x] UI Tests
- [x] Snapshot Tests (iOS only)
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by mjohnson12 — APPROVED** [review]

**Comment by mjohnson12** [comment]
You could probably follow this up with removing the @availability on the iOS 13+ code.

**Review by lickel — COMMENTED** [review]
> note: this change will require a major version bump for the next release that includes it

Would you re-adjust things such that concurrency's version == workflow again?

Also, have you considered throwing more stuff in version.rb for maintenance? Eg the target OSes/swift version

**Comment by jamieQ** [comment]
> Would you re-adjust things such that concurrency's version == workflow again?

yep! that's the plan anyway. also have some additional primary associated types to add

> Also, have you considered throwing more stuff in version.rb for maintenance? Eg the target OSes/swift version

that's a good suggestion – i'll look into it!

**Comment by lickel** [comment]
> i'll look into it!

Market uses that pattern if you want inspiration

File: Samples/ModalContainer/ModalContainer.podspec [hunk 1]
```diff
@@ -10,8 +10,8 @@ Pod::Spec.new do |s|
   # 1.7 is needed for `swift_versions` support
   s.cocoapods_version = '>= 1.7.0' 
 
-  s.swift_versions = ['5.0']
-  s.ios.deployment_target = '11.0'
+  s.swift_versions = [WORKFLOW_SWIFT_VERSION]
+  s.ios.deployment_target = WORKFLOW_IOS_DEPLOYMENT_TARGET
```
Comment (JaviSoto) [inline Samples/ModalContainer/ModalContainer.podspec:14]: I love this change instead of repeating the value everywhere 👍

**Review by JaviSoto — APPROVED** [review]

**Pull request merged**